woocommerce/woocommerce-gutenberg-products-block
# Product Gallery block: Migrate to new Interactivity API store

Fixes #11656 

## What

Migrates the Product Gallery block and its inner blocks (Pager, Thumbnails, Large Image blocks) to the new store() API.

Resources:

- https://github.com/woocommerce/woocommerce-blocks/issues/11065
- https://github.com/woocommerce/woocommerce-blocks/pull/11071
- https://github.com/woocommerce/woocommerce-blocks/pull/11558
- https://github.com/WordPress/gutenberg/discussions/53586

## Why



Because we are switching from the old `store()` API to the new `store()` API, and we need to migrate all the blocks at once.

## Testing Instructions



1. From your WordPress dashboard, go to Appearance > Themes. Make sure you have a block-based theme installed and activated. If not, you can install one from the Add New option. Block-based themes include "Twenty-twenty Three," "Twenty-twenty Four," etc.
2. On the left-hand side menu, click on Appearance > Editor > Templates
3. Find and select the 'Single Product' template from the list.
4. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
5. Inside the Page editor, click on the '+' button to add a new block.
6. In the block library that pops up, search for the 'Product Gallery' block. Click on it to add the block to the template.
7. Click on Save;
8. Visit a product page and make sure the block and its inner blocks are working as expected.

## WooCommerce Visibility


Required:


-------
